frontend/src/metabase/dashboard/components/DashboardEmbedAction/DashboardEmbedAction.tsx

```diff
@@ -0,0 +1,116 @@
+import { useState } from "react";
+import type { Dashboard } from "metabase-types/api";
+import { DashboardEmbedHeaderButton } from "metabase/dashboard/components/DashboardEmbedHeaderButton";
+import { DashboardEmbedHeaderMenu } from "metabase/dashboard/components/DashboardEmbedHeaderMenu";
+import DashboardSharingEmbeddingModal from "metabase/dashboard/containers/DashboardSharingEmbeddingModal";
+import { useSelector } from "metabase/lib/redux";
+import { getSetting } from "metabase/selectors/settings";
+import { getUserIsAdmin } from "metabase/selectors/user";
+
+export type EmbedButtonClickBehavior =
```

Comment:
Yeah I feeling a bit weird about what to name this. The reason I didn't go with `Action` was that I was worried it would be confused with redux actions, while `ClickBehavior` would be _slightly_ more accurate since we're saying "this is what happens when you click the button." Let me think about it just a little bit more

Reply:
Ok we solved this, we'll use `EmbedMenuModes`